runtime/parachains/src/hrmp.rs

```diff
@@ -534,31 +606,49 @@ impl<T: Config> Pallet<T> {
 	pub(crate) fn initializer_on_new_session(
 		notification: &initializer::SessionChangeNotification<T::BlockNumber>,
 		outgoing_paras: &[ParaId],
-	) {
-		Self::perform_outgoing_para_cleanup(&notification.prev_config, outgoing_paras);
+	) -> Weight {
+		let w1 = Self::perform_outgoing_para_cleanup(&notification.prev_config, outgoing_paras);
 		Self::process_hrmp_open_channel_requests(&notification.prev_config);
 		Self::process_hrmp_close_channel_requests();
+		w1.saturating_add(<T as Config>::WeightInfo::force_process_hrmp_open(
+			outgoing_paras.len() as u32
+		))
+		.saturating_add(<T as Config>::WeightInfo::force_process_hrmp_close(
+			outgoing_paras.len() as u32
+		))
 	}
 
 	/// Iterate over all paras that were noted for offboarding and remove all the data
 	/// associated with them.
 	fn perform_outgoing_para_cleanup(
 		config: &HostConfiguration<T::BlockNumber>,
 		outgoing: &[ParaId],
-	) {
-		Self::clean_open_channel_requests(config, outgoing);
+	) -> Weight {
+		let mut w = Self::clean_open_channel_requests(config, outgoing);
 		for outgoing_para in outgoing {
 			Self::clean_hrmp_after_outgoing(outgoing_para);
+
+			// we need a few extra bits of data to weigh this -- all of this is read internally
+			// anyways, so no overhead.
+			let ingress_count =
+				<Self as Store>::HrmpIngressChannelsIndex::get(outgoing_para).len() as u32;
+			let egress_count =
+				<Self as Store>::HrmpEgressChannelsIndex::get(outgoing_para).len() as u32;
+			w = w.saturating_add(<T as Config>::WeightInfo::force_clean_hrmp(
+				ingress_count,
+				egress_count,
+			));
 		}
+		w
 	}
 
 	// Go over the HRMP open channel requests and remove all in which offboarding paras participate.
 	//
 	// This will also perform the refunds for the counterparty if it doesn't offboard.
-	fn clean_open_channel_requests(
+	pub fn clean_open_channel_requests(
```

Comment:
pub(crate). Could better, could we do `pub(in benchmarking)`? In that case we would need to move `#[cfg(feature = "runtime-benchmarks")]` inside of the module and make it `#![..]`

Reply:
will suffice to just making pub(crate) for now. How about a `#[cfg(feature = "runtime-benchmarks")] pub(crate) fn clean_open_channel_requests_for_benchmarks` wrapper for the private one?